### PR TITLE
Map write

### PR DESCRIPTION
**Pull request by Kangz**
PTAL, this doesn't implement the wire on purpose.

### CODE REVIEW

File: src/tests/unittests/validation/BufferValidationTests.cpp [hunk 1]
```diff
@@ -184,7 +204,21 @@ TEST_F(BufferValidationTest, MapReadSuccess) {
...
 
-    EXPECT_CALL(*mockBufferMapReadCallback, Call(NXT_BUFFER_MAP_READ_STATUS_SUCCESS, Ne(nullptr), userdata))
+    EXPECT_CALL(*mockBufferMapReadCallback, Call(NXT_BUFFER_MAP_ASYNC_STATUS_SUCCESS, Ne(nullptr), userdata))
+        .Times(1);
+    queue.Submit(0, nullptr);
+
+    buf.Unmap();
+}
+
+// Test the success cause for mapping buffer for writing
```
Comment (kainino0x) [inline src/tests/unittests/validation/BufferValidationTests.cpp]: 'cause' -> 'case'?

File: src/tests/unittests/validation/BufferValidationTests.cpp [hunk 2]
```diff
@@ -270,23 +383,105 @@ TEST_F(BufferValidationTest, DISABLED_MapReadDestroyBeforeResult) {
 }
 
 // When a request is cancelled with Unmap it might still be in flight, test doing a new request
-// works as expected and we don't get the cancelled request's data.
+// works as expected and we don't get the cancelled request's data. Reading part
```
Comment (kainino0x) [inline src/tests/unittests/validation/BufferValidationTests.cpp]: Incomplete sentence? Or do you mean, like, `(Reading part.)`? In which case I would say:
```
When a MapRead is cancelled (...)
```

File: src/tests/unittests/validation/BufferValidationTests.cpp [hunk 3]
```diff
@@ -270,23 +383,105 @@ TEST_F(BufferValidationTest, DISABLED_MapReadDestroyBeforeResult) {
...
 
-    EXPECT_CALL(*mockBufferMapReadCallback, Call(NXT_BUFFER_MAP_READ_STATUS_SUCCESS, Ne(nullptr), userdata))
+    EXPECT_CALL(*mockBufferMapReadCallback, Call(NXT_BUFFER_MAP_ASYNC_STATUS_SUCCESS, Ne(nullptr), userdata))
+        .Times(1);
+    queue.Submit(0, nullptr);
+}
+// TODO(cwallez@chromium.org) Test a MapWrite and already MapRead and vice-versa
+
+// When a request is cancelled with Unmap it might still be in flight, test doing a new request
+// works as expected and we don't get the cancelled request's data. Writing part
```
Comment (kainino0x) [inline src/tests/unittests/validation/BufferValidationTests.cpp]: Here too

**Comment by kainino0x** [comment]
Needs reformatting, according to travis. Otherwise, LGTM, few nits

File: src/tests/unittests/validation/BufferValidationTests.cpp (hunk 1 above)
Comment (Kangz) [reply src/tests/unittests/validation/BufferValidationTests.cpp]: Done

File: src/tests/unittests/validation/BufferValidationTests.cpp (hunk 2 above)
Comment (Kangz) [reply src/tests/unittests/validation/BufferValidationTests.cpp]: Done

File: src/tests/unittests/validation/BufferValidationTests.cpp (hunk 3 above)
Comment (Kangz) [reply src/tests/unittests/validation/BufferValidationTests.cpp]: Done

**Comment by Kangz** [comment]
Thanks for the review!

**Pull request merged**